central/baseimage/service/service_impl.go

```diff
@@ -0,0 +1,191 @@
+package service
+
+import (
+	"context"
+	"regexp"
+
+	"github.com/distribution/reference"
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/baseimage/datastore/repository"
+	v2 "github.com/stackrox/rox/generated/api/v2"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/auth/permissions"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/pkg/grpc/authz"
+	"github.com/stackrox/rox/pkg/grpc/authz/perrpc"
+	"github.com/stackrox/rox/pkg/grpc/authz/user"
+	"github.com/stackrox/rox/pkg/images/integration"
+	imgUtils "github.com/stackrox/rox/pkg/images/utils"
+	"github.com/stackrox/rox/pkg/sac/resources"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var (
+	// TODO(ROX-32170): Review and finalize RBAC for base image repository operations.
+	authorizer = perrpc.FromMap(map[authz.Authorizer][]string{
+		user.With(permissions.View(resources.Administration)): {
+			v2.BaseImageServiceV2_GetBaseImageReferences_FullMethodName,
+		},
+		user.With(permissions.Modify(resources.Administration)): {
+			v2.BaseImageServiceV2_CreateBaseImageReference_FullMethodName,
+			v2.BaseImageServiceV2_UpdateBaseImageTagPattern_FullMethodName,
+			v2.BaseImageServiceV2_DeleteBaseImageReference_FullMethodName,
+		},
+	})
+)
+
+// serviceImpl provides APIs for base image references.
+type serviceImpl struct {
+	v2.UnimplementedBaseImageServiceV2Server
+
+	datastore      repository.DataStore
+	integrationSet integration.Set
+}
+
+// GetBaseImageReferences returns all base image references.
+func (s *serviceImpl) GetBaseImageReferences(ctx context.Context, _ *v2.Empty) (*v2.GetBaseImageReferenceResponse, error) {
+	repos, err := s.datastore.ListRepositories(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get base image repositories: %v", err)
```

Comment:
We have an errors pkg that is convenient to return API errors 
pkg - "github.com/pkg/errors"
example  - errors.Wrap(errox.ServerError, "msg")